### PR TITLE
Mon 6281: Service discovery email not working properly 

### PR DESCRIPTION
**Pull request by garnier-quentin**

### CODE REVIEW

File: gorgone/modules/centreon/autodiscovery/services/discovery.pm [hunk 1]
```diff
@@ -129,34 +130,33 @@ sub send_email {
...
-                                    $self->{mail_command},
-                                    $self->{mail_subject},
-                                    $self->{mail_from},
-                                    $self->{discovery}->{rules}->{$rule_id}->{contact}->{$contact_id}->{contact_email}
-                                )
-                            }
-                        ]
-                    }
-                );
+                my $smtp = Net::SMTP->new('localhost');
```
Comment (cgagnaire) [inline gorgone/modules/centreon/autodiscovery/services/discovery.pm:143]: Shouldn't we add a timeout here as default is 120 (like you said in https://github.com/centreon/centreon-autodiscovery/issues/116) ?

File: gorgone/modules/centreon/autodiscovery/services/discovery.pm [hunk 2]
```diff
@@ -129,34 +130,33 @@ sub send_email {
...
+                $smtp->mail($self->{mail_from});
+                if ($smtp->to($self->{discovery}->{rules}->{$rule_id}->{contact}->{$contact_id}->{contact_email})) {
```
Comment (cgagnaire) [inline gorgone/modules/centreon/autodiscovery/services/discovery.pm:145]: The condition on to function is enough ?

File: gorgone/modules/centreon/autodiscovery/services/discovery.pm (hunk 1 above)
Comment (garnier-quentin) [reply gorgone/modules/centreon/autodiscovery/services/discovery.pm:143]: It s an issue about command line length. Not about timeout.

**Review by cgagnaire — APPROVED** [review]

File: gorgone/modules/centreon/autodiscovery/services/discovery.pm (hunk 1 above)
Comment (cgagnaire) [reply gorgone/modules/centreon/autodiscovery/services/discovery.pm:143]: Ah yes.
Anyway I think we should map the timeout as a config parameter.

File: gorgone/modules/centreon/autodiscovery/services/discovery.pm (hunk 2 above)
Comment (garnier-quentin) [reply gorgone/modules/centreon/autodiscovery/services/discovery.pm:145]: Maybe i ll try with postfix off. You re right.

**Pull request merged**